lms/static/scripts/frontend_apps/utils/test/onedrive-picker-client-test.js

```diff
@@ -0,0 +1,145 @@
+import { PickerCanceledError } from '../google-picker-client';
+import { OneDrivePickerClient, $imports } from '../onedrive-picker-client';
+
+describe('OneDrivePickerClient', () => {
+  let fakeLoadOneDriveAPI;
+  const clientOptions = {
+    clientId: '12345',
+    redirectURI: 'https://redirect.uri',
+  };
+
+  function createClient({ clientId, redirectURI } = clientOptions) {
+    return new OneDrivePickerClient({ clientId, redirectURI });
+  }
+
+  beforeEach(() => {
+    fakeLoadOneDriveAPI = sinon.stub();
+
+    $imports.$mock({
+      './onedrive-api-client': {
+        loadOneDriveAPI: fakeLoadOneDriveAPI,
+      },
+    });
+    delete window.OneDrive;
```

Comment:
If you implement my suggestion in in `onedrive-picker-client.js` to have this class not directly reference `window.OneDrive` then you can more easily mock the `window.OneDrive` interface in the tests by making `fakeLoadOneDriveAPI` return a stub.

Reply:
OK, let me try it.

Reply:
This is a great suggestion, I have implemented in an added commit.